NewRelicBrowser.php

```diff
@@ -1,392 +1,456 @@
-						add_settings_error( 'relic_options', 'relic_options_error', $app_json_data->error->title, 'error' );
-						return false;
-				} else {
-						/* stored the received browser application data */
-
-						$browser_details_array = array(
-							'relic_app_name' => $app_json_data->browser_application->name,
-							'relic_app_key' => $app_json_data->browser_application->browser_monitoring_key,
-							'relic_app_id' => $app_json_data->browser_application->id,
-							'relic_app_script' => $app_json_data->browser_application->loader_script,
-						);
-						add_option( 'rtp_relic_browser_details', $browser_details_array );
-						return true;
-				}
+			$app_json_data = json_decode( $app_response['body'] );
+			if ( 201 !== $app_response_code ) {
+					add_settings_error( 'relic_options', 'relic_options_error', $app_json_data->error->title, 'error' );
+					return false;
+			} else {
+					/* stored the received browser application data */
+
+					$browser_details_array = array(
+						'relic_app_name'   => $app_json_data->browser_application->name,
+						'relic_app_key'    => $app_json_data->browser_application->browser_monitoring_key,
+						'relic_app_id'     => $app_json_data->browser_application->id,
+						'relic_app_script' => $app_json_data->browser_application->loader_script,
+					);
+					add_option( 'rtp_relic_browser_details', $browser_details_array );
+					return true;
+			}
 		}
 
-		function rtp_relic_options_validate( $input )
-		{
-
-			/* validate form if js not worked */
-			$rtp_relic_form_validated = $this->rtp_relic_validate_form( $_POST );
-			if ( ! $rtp_relic_form_validated['valid'] ) {
-				/* form not validated throw an error */
-				add_settings_error( 'relic_options', 'relic_options_error', $rtp_relic_form_validated['message'], 'error' );
-			} else {
-				/* form validated go ahead */
-				$option_name = 'rtp_relic_account_details';
-				$app_option_name = 'rtp_relic_browser_details';
-				$browser_app_list_option = 'rtp_relic_browser_list';
-
-				if ( 'rtp-select-browser' == $_POST['rtp-relic-form-name'] ) {
-					/* user has selected the application so fetch the browser script */
-					$rtp_account_details = get_option( $option_name );
-					if ( isset( $_POST['rtp-selected-browser-id'] ) ) {
-						/* In this curl we are filtering browser applications by using browser id */
-
-						$browser_list_response = wp_remote_get( RTP_NEW_RELIC_BROWSER_URL.'?filter[ids]=' . sanitize_text_field( $_POST['rtp-selected-browser-id'] ), array(
-							'timeout' => 100,
-							'headers' => array( 'x-api-key' => $rtp_account_details['relic_api_key'], 'Content-Type' => 'application/json' ),
+		/**
+		 * Relic options validate.
+		 *
+		 * @param string $input input parameter.
+		 *
+		 * @return string input.
+		 */
+		public function rtp_relic_options_validate( $input ) {
+			if ( isset( $_POST['submit_new_relic'] ) && wp_verify_nonce( sanitize_key( $_POST['submit_new_relic'] ), 'relic_options_nonce_action' ) ) {
```

Comment:
Do early return to avoid large `if` block. If nonce not verified then return.

Reply:
Change `submit_new_relic` to `relic_options_nonce`

Reply:
i solved this issue. please review changes.